src/models/connectionStore.ts

```diff
@@ -324,13 +329,21 @@ export class ConnectionStore {
         return this.doSavePassword(profile, CredentialsQuickPickItemType.Profile);
     }
 
-    private doSavePassword(conn: IConnectionCredentials, type: CredentialsQuickPickItemType): Promise<boolean> {
+    public saveProfileWithConnectionString(profile: IConnectionProfile): Promise<boolean> {
+        if (!profile.connectionString) {
+            return Promise.resolve(true);
+        }
+        return this.doSavePassword(profile, CredentialsQuickPickItemType.Profile, true);
+    }
+
+    private doSavePassword(conn: IConnectionCredentials, type: CredentialsQuickPickItemType, isConnectionString: boolean = false): Promise<boolean> {
         let self = this;
+        let password = isConnectionString ? conn.connectionString : conn.password;
```

Comment:
When you make changes like this it would really be great if you also updated the function and variable names to properly reflect the changes. For example this is no longer specifically for a password so I'd say something like doSaveCredential is more appropriate and will reduce the chance someone misinterprets what this method does in the future.